suite/runsuite.cmake

```diff
@@ -262,53 +262,57 @@ endif ()
+if (DEFINED ENV{DISABLE_CLANG_FORMAT_CHECKS} AND "$ENV{DISABLE_CLANG_FORMAT_CHECKS}" STREQUAL "yes")
+  message("clang-format check disabled")
+else ()
+  find_program(CLANG_FORMAT_DIFF clang-format-diff-14 DOC "clang-format-diff")
+  if (NOT CLANG_FORMAT_DIFF)
+    find_program(CLANG_FORMAT_DIFF clang-format-diff DOC "clang-format-diff")
   endif ()
-  if (format_out)
-    # The WARNING and FATAL_ERROR message types try to format the diff and it
-    # looks bad w/ extra newlines, so we use STATUS for a more verbatim printout.
-    message(STATUS
-      "Changes are not formatted properly:\n${format_out}")
-    message(FATAL_ERROR
-      "FATAL ERROR: Changes are not formatted properly (see diff above)!")
-  else ()
-    message("clang-format check passed")
+  if (NOT CLANG_FORMAT_DIFF)
+    find_program(CLANG_FORMAT_DIFF clang-format-diff.py DOC "clang-format-diff")
   endif ()
-else ()
-  if (arg_require_format)
-    message(FATAL_ERROR "FATAL ERROR: clang-format is required but not found!")
+  find_package(Python3)
+  if (CLANG_FORMAT_DIFF AND Python3_FOUND)
+    get_filename_component(CUR_DIR "." ABSOLUTE)
+    set(diff_file "${CUR_DIR}/runsuite_diff.patch")
+    file(WRITE ${diff_file} "${diff_contents}")
+    execute_process(COMMAND ${Python3_EXECUTABLE} ${CLANG_FORMAT_DIFF} -p1
+      WORKING_DIRECTORY "${CTEST_SOURCE_DIRECTORY}"
+      INPUT_FILE ${diff_file}
+      RESULT_VARIABLE format_result
+      ERROR_VARIABLE format_err
+      OUTPUT_VARIABLE format_out)
+    if (format_result OR format_err)
+      message(FATAL_ERROR
+        "Error (${format_result}) running clang-format-diff: ${format_err}")
+    endif ()
+    if (format_out)
+      # The WARNING and FATAL_ERROR message types try to format the diff and it
+      # looks bad w/ extra newlines, so we use STATUS for a more verbatim printout.
+      message(STATUS
+        "Changes are not formatted properly:\n${format_out}")
+      message(FATAL_ERROR
+        "FATAL ERROR: Changes are not formatted properly (see diff above)!")
+    else ()
+      message("clang-format check passed")
+    endif ()
   else ()
-    message("clang-format-diff not found: skipping format checks")
+    if (arg_require_format)
+      message(FATAL_ERROR "FATAL ERROR: clang-format is required but not found!")
+    else ()
+      message("clang-format-diff not found: skipping format checks")
+    endif ()
   endif ()
-endif ()
 
-# Check for tabs other than on the revision lines.
-# The clang-format check will now find these in C files, but not non-C files.
-string(REGEX REPLACE "\n(---|\\+\\+\\+)[^\n]*\t" "" diff_notabs "${diff_contents}")
-string(REGEX MATCH "\t" match "${diff_notabs}")
-if (NOT "${match}" STREQUAL "")
-  string(REGEX MATCH "\n[^\n]*\t[^\n]*" match "${diff_notabs}")
-  message(FATAL_ERROR "ERROR: diff contains tabs: ${match}")
+  # Check for tabs other than on the revision lines.
+  # The clang-format check will now find these in C files, but not non-C files.
```

Comment:
This is also not clang-format: so removing "clang format" from the names seems more appealing.